reader.go

```diff
@@ -543,11 +543,7 @@ func decodeLineOfMediaPlaylist(p *MediaPlaylist, wv *WV, state *decodingState, l
 		// If EXT-X-KEY appeared before reference to segment (EXTINF) then it linked to this segment
 		if state.tagKey {
 			p.Segments[p.last()].Key = &Key{state.xkey.Method, state.xkey.URI, state.xkey.IV, state.xkey.Keyformat, state.xkey.Keyformatversions}
-			// First EXT-X-KEY may appeared in the header of the playlist and linked to first segment
-			// but for convenient playlist generation it also linked as default playlist key
-			if p.Key == nil {
-				p.Key = state.xkey
```

Comment:
I think that by just not setting p.Key and let it be nil, you don't need to remove the reset of the code using p.Key, so maybe you can just add some condition here to not set it?

Reply:
I was considering that. Felt that this would leave dead code. If we go that way, change rendering only, the code would be more descriptive if we still set the pointer and change the rendering only.